packages/core/src/shared/telemetry/spans.ts

```diff
@@ -113,7 +148,17 @@ export class TelemetrySpan<T extends MetricBase = MetricBase> {
      */
     static readonly #excludedFields = ['passive', 'value']
 
-    public constructor(public readonly name: string) {}
+    public constructor(
+        public readonly name: string,
+        options?: SpanOptions
+    ) {
+        // set defaults on undefined options
+        this.#options = {
```

Comment:
can we add some tests for the emit option as well

Reply:
done